### DISCUSSION TITLE
Why ESP32 restarts every hour at minute 00?

### DISCUSSION THREAD

**Discussion by radudorurus** [arendst/Tasmota#17713] [Q&A] [answered]
I start this as a question.

I have a setup with 10 sonoff basic and one ESP32Wroom.
All communicating with HomeAssistant server, MQTT.

I observed few times the ESP32 buttons being NA on the dashboard.

I cannot say for sure if this behavior is related to HA or not.
However previous fails to load the ESP32 webpage (prior to MQTT server installation) may have been because of this behavior.

I waited for the time to reach 00 minutes having the console page open.
No MQTT message came from the HA server (related to restart).
The info page shows reason of restart: software reset CPU.
The console shows the boot sequence, connecting and in ~13 seconds the NTP sets the correct time.
Version 12.2.0

Basically the ESP has quite the same settings with the sonoffs but the sonoffs does not reset regularly.

Any ideas?

**Comment by barbudor** [comment]
There is no reason why the ESP32 would restart like this

Please set Weblog to 4 and get the logs before the restart. It should give hints on the reason of the reboot

If not, please provide more details
- What is the device ? dev board or commercial device ?
- What is the configuration ? GPIO ? Rules ? Using Bluetooth ?

**Reply by sfromis** [reply]
Something happening at 00 (may not be on the ESP32 itself) might interfere, potentially triggering a restart. Since you watch MQTT, you could also set `MqttLog 4` to have as many messages as possible visible, similar to what you'd have in the console, where you may not see it all before restart wipes out the page.

**Comment by radudorurus** [comment, marked as answer]
Thanks for the hints friends.
I think I found the culprit...
There is a RULE.

22:53:36.051 CMD: weblog
22:53:36.061 MQT: stat/ESP32PowerDiverter/RESULT = {"WebLog":2}
22:55:16.872 MQT: tele/ESP32PowerDiverter/STATE = {"Time":"2023-01-15T22:55:16","Uptime":"0T00:55:14","UptimeSec":3314,"Heap":144,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":43},"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","Wifi":{"AP":1,"SSId":"on_air","BSSId":"xxxxxxxxxxxxxx","Channel":11,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:07"}}
22:55:16.890 MQT: tele/ESP32PowerDiverter/SENSOR = {"Time":"2023-01-15T22:55:16","ESP32":{"Temperature":40.6},"TempUnit":"C"}
22:59:16.061 CMD: weblog 4
22:59:16.074 MQT: stat/ESP32PowerDiverter/RESULT = {"WebLog":4}
22:59:16.950 CFG: Saved, Count 6583, Bytes 4096
22:59:30.881 WIF: Checking connection...
22:59:42.133 CMD: mqttlog 4
22:59:42.136 SRC: WebConsole from [ip deleted]
22:59:42.140 CMD: Grp 0, Cmd 'MQTTLOG', Idx 1, Len 1, Pld 4, Data '4'
22:59:42.152 MQT: stat/ESP32PowerDiverter/RESULT = {"MqttLog":4}
22:59:42.895 CFG: Saved, Count 6584, Bytes 4096
22:59:50.880 WIF: Checking connection...
23:00:00.845 RUL: TIME#MINUTE|60 performs "RESTART 1"
23:00:00.847 SRC: Rule
23:00:00.850 CMD: Grp 0, Cmd 'RESTART', Idx 1, Len 1, Pld 1, Data '1'
23:00:00.861 MQT: stat/ESP32PowerDiverter/RESULT = {"Restart":"Restarting"}

**Reply by barbudor** [reply]
OMG
🙄

**Reply by sfromis** [reply]
![image](https://user-images.githubusercontent.com/47082390/212567721-5ea68cca-9fbe-4605-a8c0-a26f43171293.png)

**Comment by radudorurus** [comment]
Who put that rule there?
It was not me for sure.

**Reply by sfromis** [reply]
Well, someone may have had access, and decided to do a mild prank. Such may happen if you port-forward it to the Internet. By doing that, others have had unexpected help to change their setup.

**Reply by radudorurus** [reply]
I am pretty sure nobody has access. The http page is passworded (still it has access from outside by port fw).
Is it possible for the bin file to contain rules?
In the beginning I used custom compiled bin (all kind of features wanted...) and I saved the configuration.
Later I did update directly from web but I ported the configuration.

**Reply by sfromis** [reply]
Yes, a binary can be built to include rules. And upgrading to another binary will not remove such setup, unless you reset it all. And if you save a configuration, that should include rules, if you use a tool able to save it all.

**Comment by arendst** [comment]
Execute ``rule0`` to see what more fun is there. Perhaps this shines a better light over it.

**Reply by radudorurus** [reply]
You made me curious.
I listed all rules.
Just one existed and I deleted after I turned it off to be sure it will not happen again.

Thanks again.

**Reply by sfromis** [reply]
If you still have that saved configuration, you could check if it is included there.

Anyway, such a rule to restart on the hour does smell like someone making a prank.
